src/server/utils/response.utils.js

```diff
@@ -141,6 +159,86 @@ let handleExpandReadResponse = (res, base_version, Resource, resource_json = [])
 	res.status(200).json(results);
 };
 
+/**
+ * @description Handle a preferred response type https://www.hl7.org/fhir/http.html#return
+ * @function handlePreferredResponse
+ * @param {object} params - The parameters for this function
+ * @param {Express.response} params.res - Express response object
+ * @param {string} params.base_version - Which spec version is this request coming from
+ * @param {string} params.type - type of resource, e.g. Patient or Observation
+ * @param {object} params.results - Results from the operation
+ * @param {string} params.id - Id of the updated/created resource
+ * @param {string} params.resource_version - Version number of the updated resource
+ * @param {string} params.resourceUrl - The resource url
+ * @param {number} params.status - The status to return
+ * @param {Date} params.date - If included, will add a Last-Modified header
+ * @param {string} params.prefer - The request Prefer: string
+ * @param {object} params.operationOutcome - If the prefer type is OperationOutcome, this
+ * should be included, otherwise a default message is sent.
+ */
+const handlePreferredResponse = ({
+	res,
+	base_version,
+	type,
+	results,
+	id,
+	resource_version,
+	resourceUrl,
+	status = 201,
+	date,
+	prefer,
+	operationOutcome,
+}) => {
+	const acceptedPreferTypes = ['return=minimal', 'return=representation', 'return=OperationOutcome'];
+
+	switch (prefer) {
```

Comment:
Could you pull some of this logic out of the switch statement to shorten this code up a bit? It looks like the following code is in each case and could just be defined above it so it does not need to be redefined.

```javascript
if (resource_version) {
  res.set('Content-Location', `${path.join(resourceUrl, base_version, type, id, '_history', resource_version)}`);
  res.set('ETag', `W/"${resource_version}"`);
}

if (date) {
  res.set('Last-Modified', date.toISOString());
}
res.set('Location', `${base_version}/${type}/${id}`);
```

Reply:
Will do asap